plugin/src/main/scala-3/chisel3/internal/plugin/BundlePhase.scala

```diff
@@ -0,0 +1,227 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.internal.plugin
+
+import dotty.tools.dotc.*
+import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.ast.untpd
+import dotty.tools.dotc.ast.tpd.*
+import dotty.tools.dotc.ast.Trees
+import dotty.tools.dotc.core.Contexts.*
+import dotty.tools.dotc.core.Symbols.*
+import dotty.tools.dotc.core.Names
+import dotty.tools.dotc.core.StdNames.*
+import dotty.tools.dotc.core.Constants.Constant
+import dotty.tools.dotc.typer.TyperPhase
+import dotty.tools.dotc.plugins.{PluginPhase, StandardPlugin}
+import dotty.tools.dotc.transform.{Erasure, Pickler, PostTyper}
+import dotty.tools.dotc.core.Types
+import dotty.tools.dotc.core.Types.*
+import dotty.tools.dotc.core.Flags
+import dotty.tools.dotc.util.SourcePosition
+import dotty.tools.dotc.core.Decorators.toTermName
+import dotty.tools.dotc.core.Definitions
+
+import scala.collection.mutable
+
+object BundleHelpers {
+  def cloneTypeFull(tree: Tree)(using Context): Tree = {
+    val cloneSym =
+      requiredMethod("chisel3.reflect.DataMirror.internal.chiselTypeClone")
```

Comment:
I get the desire to factor this out into a helper, but am I right in assuming that `requiredMethod` is expensive and we don't want to be invoking every time we call this? I suspect that's why `ChiselNamingPhase` has all of the `required*` things be uninitialized vars set once `prepareForUnit`.

Same all of the other utilities below.

You could still make these utilities, but you might want to make these `required*` things be arguments to the functions.

Reply:
After much benchmarking of these "required" methods from dotty and looking into their implementation in the compiler it's safe to conclude that these method calls are cached and won't be helped much by refactoring. For example, here's a snippet from benchmarking of`requiredModule` (the heaviest of the required calls)

```
requiredModule took 0.1 ms (total 0.1 ms)
requiredModule took 0.0 ms (total 0.1 ms)
requiredModule took 0.0 ms (total 0.1 ms)
requiredModule took 0.0 ms (total 0.1 ms)
```